airflow/api_fastapi/core_api/datamodels/pools.py

```diff
@@ -72,6 +72,25 @@ class PoolPatchBody(BaseModel):
 class PoolPostBody(BasePool):
     """Pool serializer for post bodies."""
 
-    pool: str = Field(alias="name")
+    pool: str = Field(alias="name", max_length=256)
     description: str | None = None
     include_deferred: bool = False
+
+
+class PoolPostBulkBody(BaseModel):
+    """Pools serializer for post bodies."""
+
+    pools: list[PoolPostBody]
+
+    @field_validator("pools", mode="after")
+    def validate_pools(cls, v: list[PoolPostBody]) -> list[PoolPostBody]:
+        pool_set = set()
+        duplicates = []
+        for pool in v:
+            if pool.pool in pool_set:
+                duplicates.append(pool.pool)
+            else:
+                pool_set.add(pool.pool)
+        if duplicates:
+            raise ValueError(f"Pool name should be unique, found duplicates: {duplicates}")
+        return v
```

Comment:
Database will handle this for us. We do not need an extra query and checking in the application code. (Just catch the duplicate entry exception from the DB)